examples/gcoap/server.c

```diff
@@ -160,6 +166,41 @@ static ssize_t _riot_board_handler(coap_pkt_t *pdu, uint8_t *buf, size_t len, co
     }
 }
 
+#if IS_USED(MODULE_GCOAP_DTLS)
+static ssize_t _creds_handler(coap_pkt_t* pdu, uint8_t *buf, size_t len, coap_request_ctx_t *ctx)
+{
+#ifdef CONFIG_DTLS_PSK
+    sock_dtls_t *gcoap_sock_dtls = gcoap_get_sock_dtls();
+    const sock_udp_ep_t *remote_ep = coap_request_ctx_get_remote_udp(ctx);
+    credman_tag_t client_tag = 0;
+    credman_type_t cred_type = 0;
+    credman_credential_t credential = {0};
+    sock_dtls_session_t dtls_session = {0};
+    sock_dtls_session_set_udp_ep(&dtls_session, remote_ep);
+    if (dsm_get_session_credential_info(gcoap_sock_dtls, &dtls_session,
+                                        &cred_type, &client_tag)) {
+        puts("gcoap_cli: could not retrieve credential information");
+        return gcoap_response(pdu, buf, len, COAP_CODE_INTERNAL_SERVER_ERROR);
+    }
+    if (credman_get(&credential, client_tag, cred_type) != CREDMAN_OK) {
+        puts("gcoap_cli: could not retrieve credman credential");
+        return gcoap_response(pdu, buf, len, COAP_CODE_INTERNAL_SERVER_ERROR);
+    }
+
+    puts("Credentials used for this connection:");
+    printf("Credman tag: %d\n", client_tag);
+    printf("Credential type: %d\n", cred_type);
+    printf("Credential ID: %s\n", (char*)credential.params.psk.id.s);
+    printf("Credential key: %s\n", (char*)credential.params.psk.key.s);
+
+    return gcoap_response(pdu, buf, len, COAP_CODE_CREATED);
+#else
+    puts("gcoap_cli: Retrieving credentials is currently only implemented for PSK");
```

Comment:
Just that I didn't get to it yet. And very likely I will need to add this later anyway.
Would it make sense to make this PR a draft for now and add it in here?

Reply:
We can hold this until you implement ECC, as it looks it shouldn't be a lot of trouble IMO.

Reply:
ECC is now implemented. I also have some code for the example that I tested this with. But I didn't push that since the discussion about the example is not resolved yet. It also makes the example a lot longer. Before, the example had nothing with any ECC key in it and I had to bring in some init function to be able to give different key pairs to the different instances.

Also, there is another issue that is not directly connected to this PR but to DTLS with ECC. Since the Makefile for tinydtls was changed it sets now the buffer size for DTLS. Now I can't, or at least I don't know how to, change the buffer size without changing it directly in the tinydtls Makefile. The solution to set `CFLAGS += -DDTLS_MAX_BUF=512` in the application Makefile worked before but not anymore. The default 128 byte for the buffer is too small for the ECC handshake. The handshake fails at the Server Key Exchange message.

Reply:
https://github.com/RIOT-OS/RIOT/pull/19892 should allow to use the cflag again